pypi/release/create/action.yaml

```diff
@@ -0,0 +1,50 @@
+name: "Create PyPI release"
+description: "Publish release artifacts from an archive to PyPI"
+
+inputs:
+    archive-name:
+        description: "Name of the archive containing the artifacts, leave blank if local"
+        default: ""
+    index-url:
+        description: "The url for the PyPI index"
+        required: true
+
+runs:
+    using: composite
+    steps:
+    -   name: "[DEBUG] Inputs"
+        shell: bash
+        run: |
+            echo "==========INPUTS=========="
+            echo archive-name : ${{ inputs.archive-name }}
+            echo index-url    : ${{ inputs.index-url }}
+
+    -   name: "Download release artifacts from `${{ inputs.archive-name }}`"
+        if: ${{ !(inputs.archive-name == '') }}
+        uses: actions/download-artifact@v4
+        with:
+            name: ${{ inputs.archive-name }}
+            path: dist/
+
+    -   name: "[DEBUG] Release artifacts"
+        shell: bash
+        run: echo $(ls ./dist)
+
+    -   name: "Publish artifacts to PyPI"
```

Comment:
This also needs an environment if we're using Trusted Publisher (which we should)